### PR TITLE
chore: setup Release Please

### PR DESCRIPTION
**Pull request by anniefu**
This creates PRs that can be merged to create GitHub Releases for each of the components of Functions Framework Java.

Every release creates two PRs:
1. A "release" PR  ([example](https://github.com/anniefu/functions-framework-java/pull/56)) that updates the `pom.xml` with the next major version number
2. Once that is merged, a "snapshot" PR ([example](https://github.com/anniefu/functions-framework-java/pull/57)) that updates the `pom.xml` with a snapshot version number for the next development iteration

The PRs need to be merged manually.

### CODE REVIEW

File: .github/.release-please-manifest.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (kappratiksha) [inline .github/.release-please-manifest.json:1]: Do we need this file since it's empty?

Comment (anniefu) [reply .github/.release-please-manifest.json:1]: Yeah, it is filled in by the release please bot as it releases new versions. I believe the convention of starting the file with a `.` is to remind people it's generated.

**Comment by jihuin** [comment]
The example release PR updated the **minor** version. Is it possible to release **patch** version every time?

**Comment by kappratiksha** [comment]
The first PR should just remove the word "SNAPSHOT" from the version and not change any numbers and the second one should update the number and add the word "SNAPSHOT", right? Or am I missing anything?

**Comment by jihuin** [comment]
> The first PR should just remove the word "SNAPSHOT" from the version and not change any numbers and the second one should update the number and add the word "SNAPSHOT", right? Or am I missing anything?

That's also my understanding.

**Comment by anniefu** [comment]
Release please uses conventional commits to determine the correct semantic version of the next release: https://github.com/googleapis/release-please#how-should-i-write-my-commits

The two example PRs in the description were formed using fake `feat:` or `fix:` commits for testing, and they are for differnet components with unrelated version numbers. Release PRs remove the "SNAPSHOT" as expected, and can also increment the semantic version as appropriate based off conventional commits.

The second "snapshot" commit will always just do a patch increment and add "-SNAPSHOT" again. The version for the next release will be determined by the next "release" PR.

"release" PRs are constantly updated by the Release Please bot based off new commits to master.

**Review by kappratiksha — APPROVED** [review]

**Review by jihuin — APPROVED** [review]

**Pull request merged**